### ISSUE TITLE
`dataclass-field-order` false positive on `@dataclass(init=False)` with a hand-written `__init__`

### ISSUE THREAD

**Issue by dsfaccini** [astral-sh/ty#3976]
hey guys, this continues the batch — earlier ones are #3461, #3460, #3257 and #3256

`ty` reports `dataclass-field-order` for a class decorated `@dataclass(init=False)`, but with `init=False` no `__init__` is generated at all (you write your own), so there is no argument ordering to violate.

Playground link: https://play.ty.dev/5e9830f1-6911-4c29-828b-8ddf48950a94

### Repro (self-contained, no external deps)

```python
from dataclasses import dataclass


@dataclass(init=False)
class ObjectOutputProcessor:
    output_type: int | None = None
    outer_typed_dict_key: str | None = None
    validator: str  # required, but with a hand-written __init__ the field order is irrelevant

    def __init__(self, output: int, validator: str) -> None:
        self.output_type = output
        self.validator = validator


processor = ObjectOutputProcessor(1, 'v')
print(processor.output_type, processor.validator)
```

Run:

```bash
uvx ty check repro.py
```

### Actual result

```text
error[dataclass-field-order]: Required field `validator` cannot be defined after fields with default values
 --> repro.py:8:5
  |
6 |     output_type: int | None = None
7 |     outer_typed_dict_key: str | None = None
8 |     validator: str  # required, but with a hand-written __init__ the field order is irrelevant
  |     ^^^^^^^^^
  |
```

### Expected result

No diagnostic. `@dataclass(init=False)` suppresses the generated `__init__`, so field declaration order imposes no constraint. The `dataclass-field-order` check should be skipped when `init=False`. (Per-field `field(init=False)` is already handled correctly — this is specifically the class-level `init=False` case.) Pyright accepts this repro.

### Notes

This comes from `pydantic-ai`'s `ObjectOutputProcessor` in `_output.py`, which uses `@dataclass(init=False)` with a hand-written `__init__`.

This looks like the same root cause already noted in astral-sh/ruff#19825 (the PR that added this diagnostic): the field-order check reads raw per-field info and doesn't account for *dataclass-level* settings, only field-level ones. The `kw_only=True` instance of that was fixed (#3115), but class-level `init=False` seems to be the same gap — per-field `field(init=False)` and `kw_only=True` are already exempt from the ordering check, and ty already models `@dataclass(init=False)` as generating no `__init__` (`C(1)` is `too-many-positional-arguments`), so the class-level `init=False` case should be exempt too.

Best!
David

**Comment by dsfaccini** [comment]
many thanks!
